py/torch_tensorrt/dynamo/_compiler.py

```diff
@@ -684,12 +678,17 @@ def compile(
     )
 
     gm = exported_program.module()
+    # Move the weights in the state_dict to CPU
     logger.debug("Input graph: " + str(gm.graph))
 
     # Apply lowering on the graph module
     gm = post_lowering(gm, settings)
     logger.debug("Lowered Input graph: " + str(gm.graph))
-
+    if offload_module_to_cpu:
+        exported_program.module().to(CPU_DEVICE)
+        logger.info(
+            "The model is offloaded to CPU during compilation. If you want to keep the model on GPU, set offload_module_to_cpu=False."
+        )
```

Comment:
Consider this message: The PyTorch model was moved to the CPU to allocate all GPU memory to TensorRT. To retain the model on the GPU, set offload_module_to_cpu=False

Reply:
Also, there's one more thing we discussed - throw a warning if we predict an oom when offload_module_to_cpu=False. This could be achieved by measuring the size of the pytorch module and available GPU memory.